lhotse/cut.py

```diff
@@ -169,7 +169,8 @@ def truncate(
             # causing trouble for users expecting matching audio tensor shapes for cuts truncated to a single duration.
             new_duration = duration
         assert new_duration > 0.0
-        assert new_start + new_duration <= self.start + self.duration + 1e-5
+        if new_start + new_duration > self.start + self.duration:
+            new_duration = self.start + self.duration - new_start
```

Comment:
Hmm, I'm not sure about this one. I originally intended to "force" the user to provide the offset and duration arguments that "physically make sense", i.e. won't ask `truncate` to exceed the Cut's bounds. You're proposing to let it pass by "correcting" the duration value that the user had provided. It seems a bit dangerous and error-prone. @danpovey what do you think? 

At the very least I would move the `assert new_duration > 0.0` after this `if` statement and add a unit test for it that demonstrates the condition when it occurs and the expected behavior.

Reply:
Maybe we can just return `None` when the current Cut was not long enough? Like what the commit https://github.com/pzelasko/lhotse/pull/64/commits/f5c166020d92ced2a0928fcd271067ddd73b4e57 does.